### PR TITLE
Microsoft cas bug

### PR DESCRIPTION
**Pull request by ChanochShayner**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
In case you are a technological partner:
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

If you are not a technological partner (i.e. PANW employee, customer or individual developer), please delete this section.

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28690

## Description
Fixing the problem - when sending parameters "severity" or "resolution status" as a filter to fetch incidents.
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ronykoz** [comment]
Add RN please

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 1]
```diff
@@ -18,13 +22,15 @@
 SEVERITY_OPTIONS = {
     'Low': 0,
     'Medium': 1,
-    'High': 2
+    'High': 2,
+    'All': [0, 1, 2]
 }
 
 RESOLUTION_STATUS_OPTIONS = {
```
Comment (ronykoz) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:29]: In the daily you said something about supporting multiple options, but here you only support all

Comment (ChanochShayner) [reply Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:29]: I added the function `params_to_filter` that handle it

File: Packs/MicrosoftCloudAppSecurity/ReleaseNotes/1_0_4.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Microsoft Cloud App Security
+- Fixed a bug in fetch incidents parameters.
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/ReleaseNotes/1_0_4.md:4]: Please try and make this more informative and use "Fixed an issue where ... "

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 3]
```diff
@@ -488,7 +492,8 @@ def calculate_fetch_start_time(last_fetch, first_fetch):
     if last_fetch is None:
         if not first_fetch:
             first_fetch = '3 days'
-        first_fetch_time = date_to_timestamp(first_fetch)
+        first_fetch_dt = parse(first_fetch).replace(tzinfo=utc)
+        first_fetch_time = int(time.mktime(first_fetch_dt.timetuple()))
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:496]:
```suggestion
        first_fetch_time = int(first_fetch_dt.timestamp())
```

This should do the trick and be more readable.

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 4]
```diff
@@ -488,7 +492,8 @@ def calculate_fetch_start_time(last_fetch, first_fetch):
...
         return first_fetch_time
     else:
         last_fetch = int(last_fetch)
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:499]:
```suggestion
        return int(last_fetch)
```

I think you have pointless lines after this - correct me if I'm wrong here

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 5]
```diff
@@ -538,16 +543,39 @@ def fetch_incidents(client, max_results, last_run, first_fetch, filters):
...
+        filters['severity'] = {'eq': SEVERITY_OPTIONS[severity[0]]}
+    else:
+        severities = []
+        for severity_option in severity:
+            severities.append(SEVERITY_OPTIONS[severity_option])
+        filters['severity'] = {'eq': severities}
+    if len(resolution_status) == 1:
+        filters['resolutionStatus'] = {'eq': RESOLUTION_STATUS_OPTIONS[resolution_status[0]]}
+    else:
+        severities = []
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:558]:
```suggestion
        resolution_statuses = []
```

The same change in this else block

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 6]
```diff
@@ -538,16 +543,39 @@ def fetch_incidents(client, max_results, last_run, first_fetch, filters):
     return next_run, incidents
 
 
+def params_to_filter(severity, resolution_status):
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:546]: Put a blank line after each if and else block for better readabillity. It is a small change but makes the code look emensly better:

```
if a:
    do 1

else:
    do 2

```

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity_test.py [hunk 7]
```diff
@@ -35,17 +35,17 @@ def get_fetch_data():
...
+        (response_alerts_data, expected_filtered_alerts),
+        (response_activities_data, expected_filtered_activities),
+        (response_files_data, expected_filtered_files),
+        (response_users_accounts_data, expected_filtered_users_accounts)
     ]
 )
-def test_args_or_params_to_filter(response_data, url_suffix, expected):
-    from MicrosoftCloudAppSecurity import args_or_params_to_filter
-    res = args_or_params_to_filter(response_data, url_suffix)
+def test_args_or_params_to_filter(response_data, expected):
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity_test.py:46]:
```suggestion
def test_args_to_filter(response_data, expected):
```

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity_test.py [hunk 8]
```diff
@@ -35,17 +35,17 @@ def get_fetch_data():
```
Comment (roysagi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity_test.py:36]: You should add a quick test for `params_to_filter`

**Review by roysagi — CHANGES_REQUESTED** [review]

**Review by roysagi — APPROVED** [review]
Looks good - just check that the fetch works as expected after the changes in timestamp functions.

**Pull request merged**